src/main/java/nextstep/courses/domain/Session.java

```diff
@@ -0,0 +1,86 @@
+package nextstep.courses.domain;
+
+import nextstep.courses.domain.field.CoverImage;
+import nextstep.courses.domain.field.Period;
+import nextstep.courses.domain.field.SessionStatus;
+import nextstep.courses.domain.field.SessionType;
+import nextstep.payments.domain.Payment;
+
+import java.time.LocalDateTime;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Session {
+    private static AtomicLong autoGenId = new AtomicLong(1L);
+
+    private Long id;
+    private Long courseId;
+    private int price;
+    private Period period;
+    private CoverImage coverImage;
+    private SessionType sessionType;
+    private SessionStatus sessionStatus;
+    private int availableSlots;
+
+    public Session(Long courseId, CoverImage coverImage, SessionType sessionType, SessionStatus sessionStatus) {
+        this.id = autoGenId.getAndIncrement();
+        this.courseId = courseId;
+        this.coverImage = coverImage;
+        this.sessionType = sessionType;
+        this.sessionStatus = sessionStatus;
+    }
+
+    public Session(Long courseId, CoverImage coverImage, SessionType sessionType, SessionStatus sessionStatus, int price, int availableSlots) {
+        if (SessionType.FREE.isFree(sessionType)) {
+            throw new IllegalArgumentException("해당 강의는 무료수업이므로 가격을 정할 수 없습니다.");
+        }
+
+        this.id = autoGenId.getAndIncrement();
+        this.courseId = courseId;
+        this.coverImage = coverImage;
+        this.sessionType = sessionType;
+        this.sessionStatus = sessionStatus;
+        this.price = price;
+        this.availableSlots = availableSlots;
+    }
+
+    public Session(Long courseId) {
+        this.id = autoGenId.getAndIncrement();
+        this.courseId = courseId;
+    }
+
+    public void registerForPaidSession(Course course, Payment payment) {
+        this.availableSlots--;
+
+        if (!isFree() && isOpen() && isAvailable() && isPaymentCorrect(payment)) {
+            Long courseId = course.addSession(this);
+            this.courseId = courseId;
+        }
+    }
+
+    public void registerForFreeSession(Course course) {
+        if (isOpen() && isFree()) {
```

Comment:
`isOpen() && isFree()` 두개의 조합에 대한 의미를 나타내는 메소드이름이 있다면, 코드를 이해하는데 좀 더 즐겁지 않을까요?
`무엇 이고 무엇` 이런것은 컴퓨터가 이해하는 언어입니다.
하지만, 코드는 사람이 보죠. 
이러한 관점에서 사람이 이해하는 메시지를 보여주면 좋지 않을까요!? 🤔 

```suggestion
        if (수강신청가능?()) {
```

Reply:
말씀하신 부분 반영하였습니다!